Makefile

```
@@ -34,6 +34,10 @@ repo-data: directories
 	curl -s -S https://data.esphome.io/release/automations.json | script/collate_automations.sh > data/automations/current.json
 	curl -s -S https://data.esphome.io/beta/automations.json | script/collate_automations.sh > data/automations/beta.json
 	curl -s -S https://data.esphome.io/dev/automations.json | script/collate_automations.sh > data/automations/next.json
+	mkdir -p data/apirefs
+	curl -s -S https://api-docs.esphome.io/tags.xml | script/format_api_tags.py > data/apirefs/current.json
+	curl -s -S https://api-docs-beta.esphome.io/tags.xml | script/format_api_tags.py > data/apirefs/beta.json
+	curl -s -S https://api-docs-dev.esphome.io/tags.xml | script/format_api_tags.py > data/apirefs/next.json
```

Note: This won't work until esphome/esphome#10794 lands and is deployed everywhere, so for now I have been applying that locally, running `doxygen`, and changing this to something like:

```suggestion
	cat /path/to/esphome/esphome/api-docs/tags.xml | script/format_api_tags.py > data/apirefs/current.json
	cp data/apirefs/current.json data/apirefs/beta.json
	cp data/apirefs/current.json data/apirefs/next.json
```